cypress/e2e/downloads.cy.ts

```diff
@@ -0,0 +1,369 @@
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay-button"]').should('exist');
+        });
+      });
+
+      it('should show overlay on click', () => {
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+          cy.get('[data-testid="download-overlay-button"]').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('exist');
+        });
+      });
+    });
+
+    it('should show overlay on click', () => {
+      return loadPlayerAndSetMedia().then(() => {
+        cy.get('[data-testid="download-overlay"]').should('not.exist');
+        cy.get('[data-testid="download-overlay-button"]').click({force: true});
+        cy.get('[data-testid="download-overlay"]').should('exist');
+      });
+    });
+  });
+
+  describe('download overlay', () => {
+    describe('close button', () => {
+      beforeEach(() => {
+        cy.intercept('**/playManifest/**', {fixture: 'video.mp4'});
+      });
+
+      it('should hide overlay on click', () => {
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+          cy.get('[data-testid="download-overlay-button"]').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('exist');
+          cy.get('[data-testid="download-overlay-close-button"]').should('exist').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+        });
+      });
+    });
+
+    describe('download button', () => {
+      beforeEach(() => {
+        cy.intercept('**/playManifest/**', {fixture: 'video.mp4'}).as('playManifest');
+      });
+
+      it('should hide overlay on click', () => {
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+          cy.get('[data-testid="download-overlay-button"]').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('exist');
+          cy.get('[data-testid="download-overlay-download-button"]').should('exist').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+        });
+      });
+
+      it('should start download on click', () => {
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+          cy.get('[data-testid="download-overlay-button"]').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('exist');
+          cy.get('[data-testid="download-overlay-download-button"]').should('exist').click({force: true});
+
+          setTimeout(() => {
+            const downloadsFolder = Cypress.config('downloadsFolder');
+            const downloadPath = `${downloadsFolder}/download.mp4`;
+            cy.readFile(downloadPath);
```

Comment:
what sense of `setTimeout` block? The content of file read but haven't compared with expected value.

Reply:
[cy.readFile will fail if the file is not found](https://docs.cypress.io/api/commands/readfile#Default-file-existence-assertion)